cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SupportBundleHttpHandler.java

```diff
@@ -0,0 +1,610 @@
+    this.store = store;
+    this.namespaceQueryAdmin = namespaceQueryAdmin;
+    this.applicationLifecycleService = applicationLifecycleService;
+    this.metricsQueryHelper = metricsQueryHelper;
+    this.logReader = logReader;
+    this.logPattern =
+        cConfig.get(LoggingConfiguration.LOG_PATTERN, LoggingConfiguration.DEFAULT_LOG_PATTERN);
+  }
+
+  /** Waits for each thread to end */
+  public static void joinThread(Thread thread) {
+    try {
+      thread.join();
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Generate the Support Bundle if valid application id, workflow id, and runid are provided.
+   *
+   * @param namespaceId the namespace id
+   * @param appId the app id
+   * @param workflowName the workflow name
+   * @param runId the runid of the workflow
+   * @param needPipelineLog whether the customer need pipeline log file or not
+   * @param needSystemLog whether the customer need system log(appfabric) file or not return the
+   *     uuid of this support bundle
+   * @throws NotFoundException is thrown when the application, workflow, or runid is not found
+   */
+  @POST
+  @Path("/support/bundle")
+  public void createSupportBundle(
+      HttpRequest request,
+      HttpResponder responder,
+      @Nullable @QueryParam("namespace-id") String namespaceId,
+      @Nullable @QueryParam("app-id") String appId,
+      @Nullable @QueryParam("workflow-name") String workflowName,
+      @Nullable @QueryParam("run-id") String runId,
+      @DefaultValue("false") @QueryParam("need-pipeline-log") Boolean needPipelineLog,
+      @DefaultValue("false") @QueryParam("need-system-log") Boolean needSystemLog)
+      throws Exception {
+    UUID uuid =
+        generateSupportBundle(
+            namespaceId, appId, workflowName, runId, needPipelineLog, needSystemLog);
+    responder.sendString(
+        HttpResponseStatus.OK, String.format("Support Bundle %s generated.", uuid));
+  }
+
+  /** Generates Support Bundle for the given parameters. */
+  private UUID generateSupportBundle(
+      String namespaceId,
+      String appId,
+      String workflowName,
+      String runId,
+      Boolean needPipelineLog,
+      Boolean needSystemLog)
+      throws Exception {
+    // Generate a universal unique id for each bundle
+    if (namespaceId != null) {
+      NamespaceId namespace = new NamespaceId(namespaceId);
+      if (!namespaceQueryAdmin.exists(namespace)) {
+        throw new NamespaceNotFoundException(namespace);
+      }
+    }
+    UUID uuid = UUID.randomUUID();
+    ProgramRunId latestProgramRunId = null;
+    RunRecordDetail latestRunRecordDetail = null;
+    if (workflowName == null) {
+      workflowName = "DataPipelineWorkflow";
+    }
+    List<Thread> threadList = new ArrayList<>();
+    List<String> namespaceList = new ArrayList<>();
+    // Puts all the files under the uuid path
+    String homePath = System.getProperty("user.home") + "/support/bundle";
```

Comment:
Please have it configured through proper configuration object (e.g. `CConfiguration`)